reactDashboard/src/components/ChartDropdown/ChartDropdown.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import "./ChartDropdown.css";
+import "bootstrap/dist/css/bootstrap.min.css";
+import Dropdown from "react-bootstrap/Dropdown";
+import "../../common.css";
+import {
+  PERCENTAGE_CASES,
+  DAILY_CASES,
+  DAILY_DEATHS,
+  TOTAL_CASES,
+  TOTAL_DEATHS,
+} from "../DataCharts/DataChartsConsts";
+
+const ChartDropdown = ({
+  chartType = DAILY_CASES,
+  setChartType,
+  showPercentageTests = true,
+}) => {
+  function getChartTypeText(chartType) {
+    let chartTypeText = "";
+
+    switch (chartType) {
+      case "dailyCases":
+        chartTypeText = "Daily Cases";
+        break;
+      case "totalCases":
+        chartTypeText = "Total Cases";
+        break;
+      case "dailyDeaths":
+        chartTypeText = "Daily Deaths";
+        break;
+      case "totalDeaths":
+        chartTypeText = "Total Deaths";
+        break;
+      case "percentCases":
+        chartTypeText = "% Positive Cases";
+        break;
+      default:
+        throw new Error("Unrecognised chartType: " + chartType);
+    }
+    return chartTypeText;
+  }
+
+  return (
+    <Dropdown onSelect={(eventKey) => setChartType(eventKey)}>
+      <Dropdown.Toggle
+        className="selected-chart"
+        value={getChartTypeText(chartType)}
+        title={getChartTypeText(chartType)}
+      >
+        {chartType == null ? "Select a chart" : getChartTypeText(chartType)}
```

Comment:
Is it valid to ever have chartType null ? It can never be undefined as it defaults to DAILY_CASES above. I'd simplify it by having it throw an error if null is passed in (and have a test case to prove it)

Reply:
I have removed line 51 - I think the logic within getChartTypeText(chartType) captures this and has been tested